### ISSUE TITLE
Change recipient ordering so that it is configured per context

### ISSUE THREAD

**Issue by NSkelsey** [globaleaks/GlobaLeaks#2113]
**Current behavior**

Currently, the ordering of recipients as they are presented to whistleblowers in different contexts is global. This means the ordering can only be done once and it in some instances this has caused some pain.

**Steps to reproduce the problem or feature illustration**

The ordering of recipients should be per context and requires the move of the `presentation_order` column off of `receiver` and onto `receiver_context` and a similar move of UI from Recipient configuration to Context Configuration.

**What is the motivation or use case for changing the behavior?**

The system is cleaner with a per context ordering of recipients.

**Comment by NSkelsey** [comment]
Okay the commit c4602a2 on branch `refactor/2113` contains the angular and htmls changes to provide ordering functionality in the frontend. The backend still needs:

1) A move of `presentation_order` to the `receiver_context` table
2) Assignment of `presentation_order` on POST and PUT for a specific context
3) Ordering of context.receiver during serialization based on the assigned order.

This is what the current UI looks like this:
![current-ui](https://user-images.githubusercontent.com/1095794/31716509-7a62492a-b408-11e7-9722-ccdb71f6aec2.png)


The end-2-end test `test-admin-context.js` needs to be updated as well. The testing only every clicked `select_all_receivers` which in this case provides no coverage ordering functionality.

**Comment by NSkelsey** [comment]
The refactor/2113 branch grew a fair amount to handle the reordering of steps, contexts and field options.

With @evilaliv3 we looked at if it was reasonable to also use the kind of logic we applied on these interfaces also for ordering fields in `fieldgroups` and when they are attached to steps. The interface, the amount of changes required and the testing needed make it not worth pursuing.

This is what the positioning interface for fields looks like now:

![ordering-with-x-y](https://user-images.githubusercontent.com/1095794/31885006-5a67cdb2-b7ef-11e7-9930-25e0d0071166.png)


With the output in /submission that produces reasonable results:
![output-submission](https://user-images.githubusercontent.com/1095794/31885012-63de201c-b7ef-11e7-9d54-41e9194ccfc9.png)
